apps/ejabberd/src/mod_roster.erl

```diff
@@ -397,10 +410,23 @@ do_process_item_set(JID1,
                     To,
                     #xmlel{attrs = Attrs, children = Els}) ->
     LJID = jid:to_lower(JID1),
+    Item = get_roster_by_jid(LUser, LServer, LJID),
```

Comment:
Why this roster `get` and `set` was moved out of transaction? In fact there 2 separate transactions now (if I read it correctly) one for `get`, the other for `set`. Things might have changed between this 2 transactions. We either need to get back to single transaction completely get rid of these transactions as in current state they doesn't make any sense to me.
I may be missing sth so please correct me where my reasoning is wrong.